torch/distributed/_pycute/int_tuple.py

```diff
@@ -39,72 +36,77 @@
 
 from functools import reduce
 from itertools import chain
-from typing import Union
+from typing import Optional, Union
+from typing_extensions import TypeAlias, TypeGuard
 
 from .typing import Integer
 
 
-def is_int(x):
+# Type aliases for better readability
+IntTuple: TypeAlias = Union[int, tuple["IntTuple", ...]]
+
+
+def is_int(x: object) -> TypeGuard[int]:
```

Comment:
```suggestion
def is_int(x: object) -> TypeIs[int]:

Reply:
Hmm I think as I have explained, TypeIS is not available in python3.9 while we have CI for that. So we won't take your comment for now. And down the road, if we bump python version for PT, We will definitely use TypeIs. Does that sound good?

Reply:
@fduwjj It is available through `typing_extensions`